### PR TITLE
Add regression test for job creation without configure permission ambiguity

### PR DESCRIPTION
**Pull request by Ambuj123554**
This PR adds a test to verify the behavior when a user has Job/Create
permission but does not have Job/Configure permission.

The goal is to ensure that permission ambiguity is handled correctly
during job creation and to prevent regressions in authorization logic.

### Testing done

- Added a new test case: `jobCreateWithoutConfigureIsAmbiguous`
- Verified locally using:

  mvn -Dtest=AmbiguityTest#jobCreateWithoutConfigureIsAmbiguous test

- The test passes successfully and exercises the relevant permission
  handling code path.

### Submitter checklist

- [x] Make sure you are opening from a topic/feature/bugfix branch
- [x] Ensure the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira (if applicable)
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/matrixauth/AmbiguityTest.java [hunk 1]
```diff
@@ -124,6 +126,72 @@ void usersAreUsers() throws Exception {
...
+        nameInput.setValueAttribute("ambiguous-job");
+
+        HtmlRadioButtonInput mode =
+                (HtmlRadioButtonInput) newJobPage.getElementsByName("mode").get(0);
+        mode.click();
+
+        HtmlForm form = newJobPage.getFormByName("createItem");
+        HtmlFormUtil.submit(form);
+
+        assertThrows(FailingHttpStatusCodeException.class, () -> wc.goTo("job/ambiguous-job/configure"));
```
Comment (daniel-beck) [inline src/test/java/org/jenkinsci/plugins/matrixauth/AmbiguityTest.java:155]: This assertion seems wrong, the exact opposite is expected.

File: hs_err_pid13904.log
Comment (daniel-beck) [inline hs_err_pid13904.log:1]: Why is this here? Delete.

File: hs_err_pid8828.log
Comment (daniel-beck) [inline hs_err_pid8828.log:1]: Delete.

File: src/test/java/org/jenkinsci/plugins/matrixauth/AmbiguityTest.java [hunk 2]
```diff
@@ -124,6 +126,72 @@ void usersAreUsers() throws Exception {
...
+        j.jenkins.setAuthorizationStrategy(new FullControlOnceLoggedInAuthorizationStrategy());
+    }
+
+    @AfterEach
+    void resetAuthorization() {
+        if (j == null) {
+            return;
+        }
+        j.jenkins.setAuthorizationStrategy(new ProjectMatrixAuthorizationStrategy());
+    }
```
Comment (daniel-beck) [inline src/test/java/org/jenkinsci/plugins/matrixauth/AmbiguityTest.java:193]: This looks like a mess, why are these here?

**Review by daniel-beck — CHANGES_REQUESTED** [review]
It seems this covers the behavior of project-based matrix strategy granting _Configure_ permission to the user who _Create_'d the object. That is already covered by https://github.com/jenkinsci/matrix-auth-plugin/blob/master/src/test/java/hudson/security/ProjectMatrixAuthorizationStrategyTest.java.

Please also note that "ambiguity" in the context of this plugin refers to the historical behavior of assigning permissions to any user or group of a given name, rather than just the user, or just the group. So the placement of this test makes no sense.

**Comment by Ambuj123554** [comment]
Thanks for the clarification!

I understand that this behavior is already covered in
ProjectMatrixAuthorizationStrategyTest.

I will update this PR to either cover a missing edge case
or close it if it does not add new value.

I appreciate the review.

**Comment by Ambuj123554** [comment]
Thanks for the review and clarification.

I understand that my test was based on an incorrect assumption about the expected behavior
and that this functionality is already covered elsewhere.

I will close this PR and work on a different test case that better aligns with the intended behavior.
Thank you for the guidance.
